### PR TITLE
JS: add taint-step for markdown-it when the HTML flag is set

### PR DESCRIPTION
**Pull request by erik-krogh**
Adds a taint-step for CVE-2020-27224

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/frameworks/Markdown.qll [hunk 1]
```diff
@@ -118,3 +118,20 @@ private class SnarkdownStep extends TaintTracking::SharedTaintStep {
...
 }
+
+/**
+ * A taint step for the `markdown-it` library.
+ */
+private class MarkdownItStep extends TaintTracking::SharedTaintStep {
+  override predicate step(DataFlow::Node pred, DataFlow::Node succ) {
+    exists(API::CallNode renderer, API::CallNode call |
+      renderer = API::moduleImport("markdown-it").getACall() and
+      renderer.getParameter(0).getMember("html").getARhs().asExpr().(BooleanLiteral).getValue() =
```
Comment (asgerf) [inline javascript/ql/src/semmle/javascript/frameworks/Markdown.qll:129]: Would you like to go ahead and add `DataFlow::Node.getBooleanValue`?

**Review by asgerf — DISMISSED** [review]
LGTM with an optional suggestion

File: javascript/ql/src/semmle/javascript/frameworks/Markdown.qll (hunk 1 above)
Comment (erik-krogh) [reply javascript/ql/src/semmle/javascript/frameworks/Markdown.qll]: I should have used the existing `mayHaveBooleanValue`.

**Comment by erik-krogh** [comment]
I improved the model to support a taint-step for CVE-2020-27666

File: javascript/ql/src/semmle/javascript/frameworks/Markdown.qll [hunk 2]
```diff
@@ -118,3 +118,43 @@ private class SnarkdownStep extends TaintTracking::SharedTaintStep {
...
+      call = API::moduleImport("markdown-it").getMember("Markdown").getAnInvocation()
+    |
+      call.getParameter(0).getMember("html").getARhs().mayHaveBooleanValue(true) and
+      result = call.getReturn()
+    )
+    or
+    exists(API::CallNode call |
+      call = markdownIt().getMember(["use", "set", "configure", "enable", "disable"]).getACall() and
+      result = call.getReturn() and
+      not call.getParameter(0).getARhs().getALocalSource() =
```
Comment (asgerf) [inline javascript/ql/src/semmle/javascript/frameworks/Markdown.qll:142]:
```suggestion
      not call.getParameter(0).getAValueReachingRhs() =
```
This one handles interprocedural flow

**Review by asgerf — APPROVED** [review]

**Comment by erik-krogh** [comment]
[Evaluation is fine](https://github.com/dsp-testing/erik-krogh-dca/tree/run/markdownIt-popular-security/reports).

**Pull request merged**